src/librustc_typeck/astconv.rs

```diff
@@ -630,7 +638,48 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
                                 if arg_count.correct.is_ok()
                                     && arg_count.explicit_late_bound == ExplicitLateBound::No
                                 {
-                                    Self::generic_arg_mismatch_err(tcx.sess, arg, kind.descr());
+                                    // We're going to iterate over the parameters to sort them out, and
+                                    // show that order to the user as a possible order for the parameters
+                                    let mut ordered_params = defs.params.clone();
+                                    ordered_params.sort_by(|a, b| {
```

Comment:
You should be able to use `ParamKindOrd` for this. E.g. as in
https://github.com/rust-lang/rust/blob/a59264b01247836c70e24217e0d346b868387525/src/librustc_ast_lowering/lib.rs#L893-L897

Reply:
Will do